app/unit-tests/src/org/commcare/android/mocks/ModernHttpRequesterMock.java

```diff
@@ -60,17 +78,21 @@ public static void setRequestPayloads(String[] payloadReferences) {
     }
 
     @Override
-    protected HttpURLConnection setupConnection(URL builtUrl) throws IOException {
+    public Response<ResponseBody> makeRequest() throws IOException {
+        if(isAuthenticated && !new URL(url).getProtocol().contentEquals("https")){
+            throw new AuthenticationInterceptor.PlainTextPasswordException();
+        }
+
         if (!expectedUrlStack.isEmpty()) {
-            assertUrlsEqual(expectedUrlStack.remove(0), builtUrl.toString());
+            assertUrlsEqual(expectedUrlStack.remove(0), buildUrlWithParams().toString());
         }
 
         if (requestPayloadStack.isEmpty()) {
-            return HttpURLConnectionMock.mockWithEmptyStream(builtUrl, responseCodeStack.remove(0));
+            return OkHTTPResponseMock.createResponse(responseCodeStack.remove(0));
         } else {
             String payloadReference = requestPayloadStack.remove(0);
             if (payloadReference == null) {
-                return HttpURLConnectionMock.mockWithErroringStream(builtUrl, responseCodeStack.remove(0));
+                throw new IOException(ioErrorMessage);
```

Comment:
I think this line is the wrong replacement for the line above it?

Reply:
In non retrofit code what `HttpURLConnectionMock` does is returns an `IOException` when we call `getInputStream`. Now the `makeRequest()` function we are in is responsible for executing the request and returning the response unlike earlier when `ModernHttpRequesterMock` was  returning the `HTTPConnection` which in turn was throwing this exception when `getInputStream()` used to call. So since the responsibility of `ModernHttpRequesterMock` has changed from just returning the connection to execute and returning the response, I think the above change is justified.